dynamax/hmm/models/autoregressive_hmm.py

```diff
@@ -30,6 +30,14 @@ def __init__(self,
                          initial_probs_concentration=initial_probs_concentration,
                          transition_matrix_concentration=transition_matrix_concentration)
 
+    @property
+    def emission_shape(self):
+        return (self.emission_dim,)
+
+    @property
+    def covariates_shape(self):
+        return dict(features=(self.feature_dim,))
```

Comment:
Are you implicitly assuming that the dictionary of covariates always has a key called "features"?
Why use a dictionary and not just pass in a single array?